server/src/main/java/org/apache/iotdb/db/mpp/execution/QueryStateMachine.java

```diff
@@ -125,6 +133,13 @@ public void transitionToAborted() {
     queryState.set(QueryState.ABORTED);
   }
 
+  public void transitionToFailed() {
+    if (queryState.get().isDone()) {
+      return;
+    }
+    queryState.set(QueryState.FAILED);
+  }
+
```

Comment:
Better to reuse the `public void transitionToFailed(Throwable throwable)`, you can pass a IoTDBException("retry times reach max...") or something like that.